### ISSUE TITLE
Make build process working on Windows machines

### ISSUE THREAD

**Issue by aichukanov** [hcengineering/platform#3560]
I've got a fail on `rush build` and don't know why, because of encoding problem in powershell/cmd/bash (Windows 10)

![image](https://github.com/hcengineering/anticrm/assets/6716346/3dae14c9-b952-4184-a6b4-67323f292dd3)

**Comment by aichukanov** [comment]
Found that It's because there is no `rm` on Windows. Removed using `rm -rf` from commands as a workaround.

**Comment by haiodo** [comment]
@aichukanov I think nobody was build it on windows.

**Comment by Frumcheg** [comment]
@aichukanov Hi! I manage to run project on my windows machine using WSL. You can find install guide [here](https://learn.microsoft.com/en-us/windows/wsl/install).

**Comment by CyberPanCake3000** [comment]
Hi!
@aichukanov, I was struggling with building the project on windows with WSL too but found out that visual studio reformatted shell files and files from /common/scripts directory (they are important for setting up the project) from LF to CRLF 😞 
I hope it will help you and other devs 🌟

**Comment by aeswibon** [comment]
I am also facing the same issue. Build process is failing due to `rm -r` not being present in Windows. Maybe we can use packages like `rimraf` to replace `rm -r`??

**Comment by aeswibon** [comment]
I faced multiple errors while trying to bundle the packages with `rush bundle`. I just switched to `Git bash` on Windows and replaced `mkdir -p` with `rm -rf bundle && mkdir` so that it would first delete the old dir and then create the new dir. Moreover, still facing some issues, like:

> --[ FAILURE: @hcengineering/pod-account (bundle) ]-----------[ 3.03 seconds ]--
> --minify was unexpected at this time.

> --[ FAILURE: @hcengineering/pod-front (bundle) ]-------------[ 3.38 seconds ]--
> --bundle was unexpected at this time.

> --[ FAILURE: @hcengineering/pod-server (bundle) ]------------[ 3.52 seconds ]--
> --define:process.env.GIT_REVISION was unexpected at this time.

> --[ FAILURE: @hcengineering/tool (bundle) ]------------------[ 3.28 seconds ]--
> --define:process.env.GIT_REVISION was unexpected at this time.

**Comment by Fatema-Nawrin** [comment]
@aeswibon I am facing the same issues. Did you solve it?

**Comment by aeswibon** [comment]
Nope, I tried to fix the script but is not working. Currently shifted to WSL to run the project.

**Comment by FesterCluck** [comment]
Is this actually working or are folks still stuck in WSL? I'm about to clone and give this a shot. If it's still broken, happy to pick up. Otherwise, recommend close.

**Comment by stan-kondrat** [comment]
I investigated this bug and found potential solutions:

1. **Use `cross-spawn`** in the following files to improve cross-platform compatibility:  
   - `platform-rig/bin/compile.js`  
   - `platform-rig/bin/do-svelte-check.js`  
   - `templates/apply.js`  

2. **Install Bash on Windows** and add the following to `common/config/rush/.npmrc`:  
   ```
   script-shell=bash
   ```  
   However, this approach has a blocker since **rush** doesn't support the `script-shell` config (while `npm run` does). I've created an issue in the Rush repository to address this limitation: [microsoft/rushstack#5076](https://github.com/microsoft/rushstack/issues/5076).
